### PR TITLE
Refactor amm pools tests

### PR DESCRIPTION
**Pull request by kris-ipor**

### CODE REVIEW

File: test/utils/AmmPoolUtils.sol [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        uint256 initialExchangeRate;
+        uint256 exchangeRateAfter28Days;
+        uint256 exchangeRateAfter56DaysBeforeClose;
+        int256 payoff1After28Days;
+        int256 payoff2After28Days;
+        int256 payoff1After56Days;
+        int256 payoff2After56Days;
+    }
+
+    function calculateSoap(
```
Comment (mario-ipor) [inline test/utils/AmmPoolUtils.sol:21]: Don't need this because you can reuse _iporProtocol.ammSwapsLens.getSOAP(address(_iporProtocol.asset));

Comment (kris-ipor) [reply test/utils/AmmPoolUtils.sol]: you are right, changed

File: test/amm/AmmPoolsExchangeRateLiquidity.t.sol [hunk 2]
```diff
@@ -85,85 +70,82 @@ contract JosephExchangeRateLiquidity is TestCommons, DataUtils, SwapUtils {
...
 
         vm.prank(_liquidityProvider);
-        _iporProtocol.joseph.provideLiquidity(TestConstants.TC_TOTAL_AMOUNT_10_000_18DEC);
+        _iporProtocol.ammPoolsService.provideLiquidityDai(_liquidityProvider, TestConstants.USD_10_000_18DEC);
 
         //simulation that Liquidity Pool Balance equal 0, but ipToken is not burned
-        _iporProtocol.ammStorage.setJoseph(_userOne);
-        vm.prank(_userOne);
+        AmmStorage implementationHack = new AmmStorage(_admin, address(_iporProtocol.ammTreasury));
+        _iporProtocol.ammStorage.upgradeTo(address(implementationHack));
```
Comment (mario-ipor) [inline test/amm/AmmPoolsExchangeRateLiquidity.t.sol:80]: use simple prank on router address

File: test/amm/AmmPoolsExchangeRateAndSoap.t.sol [hunk 3]
```diff
@@ -0,0 +1,491 @@
...
+        assertLt(soap, TestConstants.ZERO_INT);
+        assertLt(soap * -1, int256(balance.liquidityPool));
+        assertEq(actualExchangeRate, 1009369072035123743);
+    }
+
+    function testShouldCalculateExchangeRatePayFixedWhenSOAPChangedAndSOAPIsGreaterThanZeroAndSOAPAbsoluteValueIsLowerThanLiquidityPoolBalance()
+        public
+    {
+        // given
+        _cfg.spread28DaysTestCase = BuilderUtils.Spread28DaysTestCase.CASE0;
```
Comment (mario-ipor) [inline test/amm/AmmPoolsExchangeRateAndSoap.t.sol:104]: In old test spread for pay fixed was 4% so here you should use Spread28DaysTestCase.CASE1

_cfg.spread28DaysTestCase = BuilderUtils.Spread28DaysTestCase.CASE1;

File: test/amm/AmmPoolsExchangeRateAndSoap.t.sol [hunk 4]
```diff
@@ -0,0 +1,491 @@
...
+        _cfg.approvalsForUsers = _users;
+        _cfg.iporOracleUpdater = _userOne;
+        _cfg.iporRiskManagementOracleUpdater = _userOne;
+    }
+
+    function testShouldCalculateExchangeRatePayFixedWhenSOAPChangedAndSOAPIsLowerThanZeroAndSOAPAbsoluteValueIsLowerThanLiquidityPoolBalance()
+        public
+    {
+        // given
+        _iporProtocol = _iporProtocolFactory.getDaiInstance(_cfg);
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply test/amm/AmmPoolsExchangeRateAndSoap.t.sol:30]: done

File: test/amm/AmmPoolsExchangeRateAndSoap.t.sol (hunk 3 above)
Comment (kris-ipor) [reply test/amm/AmmPoolsExchangeRateAndSoap.t.sol]: done

File: test/amm/AmmPoolsExchangeRateAndSoap.t.sol [hunk 5]
```diff
@@ -0,0 +1,491 @@
...
+        assertEq(actualExchangeRate, 236126850257564954);
+        assertEq(soap, -8962330971951994053936);
+        assertEq(balance.liquidityPool, 5205280043501903214450);
+    }
+
+    function testShouldCalculateExchangeRateReceiveFixedWhenSOAPChangedAndSOAPIsLowerThanZeroAndSOAPAbsoluteValueIsGreaterThanLiquidityPoolBalance()
+        public
+    {
+        // given
+        _cfg.spread28DaysTestCase = BuilderUtils.Spread28DaysTestCase.CASE4;
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply test/amm/AmmPoolsExchangeRateAndSoap.t.sol]: done

File: test/amm/AmmPoolsExchangeRateAndSoap.t.sol [hunk 6]
```diff
@@ -0,0 +1,491 @@
...
+        assertEq(actualExchangeRate, 991160362216808713);
+    }
+
+    function testShouldNotCalculateExchangeRatePayFixedWhenSOAPChangedAndSOAPIsGreaterThanZeroAndSOAPAbsoluteValueIsGreaterThanLiquidityPoolBalance()
+        public
+    {
+        // given
+        _cfg.spread28DaysTestCase = BuilderUtils.Spread28DaysTestCase.CASE0;
+
+        _iporProtocol = _iporProtocolFactory.getDaiInstance(_cfg);
```
[the remark this answers is not in the document]
Comment (kris-ipor) [reply test/amm/AmmPoolsExchangeRateAndSoap.t.sol:186]: done

**Review by mario-ipor — APPROVED** [review]

**Pull request merged**